core/include/vecmem/containers/impl/device_vector.ipp

```diff
@@ -299,6 +381,34 @@ VECMEM_HOST_AND_DEVICE void device_vector<TYPE>::resize(size_type new_size,
     asize.store(new_size);
 }
 
+template <typename TYPE>
+VECMEM_HOST_AND_DEVICE void device_vector<TYPE>::resize_implicit(
+    size_type new_size) {
+    // This can only be done on a resizable vector.
+    assert(m_size != nullptr);
+
+    static_assert(details::is_implicit_lifetime_v<TYPE>,
```

Comment:
Unfortunately I forgot about testing this function. I believe this line would also produce a warning with CUDA 10.4 right now.

```suggestion
    static_assert(details::is_implicit_lifetime<TYPE>::value,
```

I should have checked for this in my previous commit. :frowning:

Reply:
What? Why? The CUDA 10.4 problem is that the `_v` predicates in the standard library were only introduced in C++17, but we define this one ourselves.